### PR TITLE
Fix few lightning issues related to the new Sunlight map

### PR DESCRIPTION
**Pull request by KurzedMetal**
#### Summary
```SUMMARY: Bugfixes "Fix few lightning issues related to the new Sunlight map"```

#### Purpose of change
Fix few issues I noticed in one of my save games:
* Underground default lighting was too high in normal conditions (like 5-6 tiles without Night Vision)
* Lighting inside houses was too high too, ~6 tiles in bright daylight in completely closed rooms, ~4 tiles at night.
* Inside Vehicles was fully affected by sunlight
* I definitely lowered a bit the default light coming from the sun inside houses and basements. I included this balance change as part of the fix because I don't seem to remember the game gave the player so much vision in those situations, I feel this is a bug too.

#### Describe the solution
I made multiple small tweaks here and there to the code, and I tried to explain the reasoning for the ones that weren't obvious. Please read diffs and commit messages.

After these changes, default vision conditions (no weather/NV/9per):
* Inside houses at night and underground vision defaults to 3 tiles radius
* Inside houses in a bright day is 4 tiles radius
* Outside is whatever moonlight / weather gives you (like always)

#### Describe alternatives you've considered
I did introduce some balancing with this bugfix, lightning inside houses and in basements were letting the player have long vision ranges. If main devs think it's a bit too much or I misremembered previous vision ranges, it can be easily reverted back changing the variable "inside_light_level" definition in build_sunlight_cache().
I did some tests in normal weather and character vision conditions and I think these values are reasonable tho.
IMO it could be even more cruel on basements and inside closed houses at night considering the player shouldn't be able able to see shit without a light source.

### CODE REVIEW

**Comment by kevingranade** [comment]
>  IMO it could be even more cruel on basements and inside closed houses at night considering the player shouldn't be able able to see shit without a light source.

I totally agree with this, IIRC it was explicitly supposed to be just enough light to grant one tile of sight in all directions since the engine can't handle 0 tiles of sight, I guess it wandered a bit since then.

```
// Just enough light that you can see the current and adjacent squares with normal vision.
#define LIGHT_AMBIENT_MINIMAL 3.7f
```

File: src/lightmap.cpp [hunk 1]
```diff
@@ -303,7 +300,7 @@ void map::generate_lightmap( const int zlev )
     const int minz = zlevels ? -OVERMAP_DEPTH : zlev;
     const int maxz = zlevels ? OVERMAP_HEIGHT : zlev;
     // Iterate top to bottom because sunlight cache needs to construct in that order.
-    for( int z = maxz; z >= minz; z-- ) {
+    for( int z = maxz; z >= zlev; z-- ) {
```
Comment (kevingranade) [inline src/lightmap.cpp]: What is the purpose of this change? I'm pretty sure it means sunlight can't penetrate lower than the level the player is standing on.

Comment (KurzedMetal) [reply src/lightmap.cpp]: My bad, for some reason I thought generate_lightmap() was being called for each zlevel. I'll revert it.

**Comment by KurzedMetal** [comment]
Just fixed the tests for these changes.
